app.rb

```diff
@@ -0,0 +1,70 @@
+require 'sinatra/base'
+require_relative 'models/user'
+require_relative 'models/peep'
+require_relative 'models/datamapper_setup'
+
+ENV['RACK_ENV'] ||= 'development'
+
+DataMapper.finalize
+DataMapper.auto_upgrade!
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+
+  get '/' do
+    redirect '/home'
+  end
+
+  get '/sign-up' do
+    erb :sign_up
+  end
+
+  post '/new-user' do
+    User.new_user_create(params[:name], params[:username], params[:email], params[:password])
+    session["username"] = params[:username] if User.redirect == '/home'
+    redirect User.redirect
+  end
+
+  get '/login' do
+    erb :login
+  end
+
+  post '/user-login' do
+    username = User.login_check(params[:username], params[:password])
+    session["username"] = username
```

Comment:
Could the lines than set session username be refactored to use a helper function eg:

```
def log_in_user(username)
  session["username"] = username
end
```

That way, the idea of how to log in a user is abstracted into one place.  This makes the mechanism of logging in a user easier to change.  It also makes it clearer what the code is doing.